canal/sync.go

```diff
@@ -142,7 +143,15 @@ func (c *Canal) runSyncBinlog() error {
 		case *replication.QueryEvent:
 			stmts, _, err := c.parser.Parse(string(e.Query), "", "")
 			if err != nil {
-				log.Errorf("parse query(%s) err %v, will skip this event", e.Query, err)
+				msg := err.Error()
+				if strings.Contains(strings.ToLower(msg), strings.ToLower("procedure")) {
```

Comment:
Parsing error message like this is a bad practice.

Don't you have an error code should could catch?